client/src/pages/EpisodeShowPage/Timer/index.tsx

```diff
@@ -0,0 +1,127 @@
+import * as _ from "lodash";
+import * as moment from "moment";
+import * as React from "react";
+import { Glyphicon } from "react-bootstrap";
+
+class TimerPresentational extends React.Component<any,  any> {
+  public render() {
+    const timerRunning = !(this.props.timeLeft === Infinity || this.props.timeLeft <= 0);
+
+    return (
+      timerRunning
+      &&
+      <span>
+        <span
+          style={{
+            color: "#137a9a",
+            fontVariant: "small-caps",
+            fontWeight: 700,
+          }}
+        >
+          <Glyphicon glyph="time" /> time left
+        </span>
+          <br />
+        <span
+          style={{
+            fontSize: "28px",
+          }}
+        >
+          {moment(this.props.timeLeft).format("m")}
+        </span>
+        m
+        {" "}
+        <span
+          style={{
+            fontSize: "28px",
+          }}
+        >
+          {moment(this.props.timeLeft).format("ss")}
+        </span>
+        s
+      </span>
+    );
+  }
+}
+
+export class Timer extends React.Component<any,  any> {
+  public state = {
+    timeLeft: Infinity,
+  };
+
+  private localStorageKey;
+  private pollingInterval;
+
+  public constructor(props: any) {
+    super(props);
+    this.localStorageKey = `mosiac-timer-${props.workspaceId}`;
```

Comment:
Sorry to say this, but I'm not sure if this is the best technique either. I would expect that this may produce lots of local storage entries that may be difficult to delete and manage.   Right now I believe that these aren't ever removed, which would be a nice addition,  but may not be enough.

I think I suggest something like having a more formal data structure for this;  have the same stuff but nested inside a key, like "mosaic". It would be more code but it would make debugging easier.

Reply:
In general I like to be extra clean & careful when dealing with local storage. It's an external entity  and is relatively important to be understandable for debugging purposes.